example/src/main/java/io/netty/example/stomp/websocket/StompWebSocketClientPageHandler.java

```diff
@@ -0,0 +1,132 @@
+ * under the License.
+ */
+package io.netty.example.stomp.websocket;
+
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.DefaultFileRegion;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.util.CharsetUtil;
+
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.net.URL;
+
+import static io.netty.handler.codec.http.HttpHeaderNames.*;
+import static io.netty.handler.codec.http.HttpHeaderValues.KEEP_ALIVE;
+import static io.netty.handler.codec.http.HttpHeaderValues.*;
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
+import static io.netty.handler.codec.http.HttpVersion.*;
+
+@Sharable
+public final class StompWebSocketClientPageHandler extends SimpleChannelInboundHandler<FullHttpRequest> {
+
+    static final StompWebSocketClientPageHandler INSTANCE = new StompWebSocketClientPageHandler();
+
+    private StompWebSocketClientPageHandler() {
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) {
+        if (request.headers().contains(HttpHeaderNames.UPGRADE, HttpHeaderValues.WEBSOCKET, true)) {
+            ctx.fireChannelRead(request.retain());
+            return;
+        }
+
+        if (!sendResource(request, ctx)) {
+            FullHttpResponse notFound = new DefaultFullHttpResponse(request.protocolVersion(), NOT_FOUND);
+            notFound.headers().set(CONTENT_TYPE, TEXT_PLAIN);
+            String payload = "Requested resource " + request.uri() + " not found";
+            notFound.content().writeCharSequence(payload, CharsetUtil.UTF_8);
+            HttpUtil.setContentLength(notFound, notFound.content().readableBytes());
+            sendResponse(notFound, ctx, true);
+        }
+    }
+
+    private static boolean sendResource(FullHttpRequest request, ChannelHandlerContext ctx) {
+        String requestResource = request.uri().substring(1);
+        if (requestResource.isEmpty()) {
+            requestResource = "index.html";
+        }
+
+        URL resourceUrl = INSTANCE.getClass().getResource(requestResource);
+        if (resourceUrl == null) {
+            return false;
+        }
+
+        RandomAccessFile raf = null;
+        long fileLength = -1L;
+        try {
+            raf = new RandomAccessFile(resourceUrl.getFile(), "r");
+            fileLength = raf.length();
+        } catch (IOException io) {
```

Comment:
Should this be `FileNotFoundException` ?

Reply:
Handled a both possible exception